content/webapp/pages/works/[workId]/items.tsx

```diff
@@ -226,7 +231,7 @@ const ItemPage: NextPage<Props> = ({
       setShowModal(false);
       setShowViewer(true);
     }
-  }, []);
+  }, [needsModal]);
```

Comment:
Can you run me through this change?

Reply:
needsModal now takes the users role into account. The role is updated client side with useUser, which means the needsModal value can also change, so we need to run this code if it does

Reply:
thank you!